appinventor/appengine/src/com/google/appinventor/client/editor/simple/palette/SimpleComponentDescriptor.java

```diff
@@ -611,8 +594,18 @@ public static MockComponent createMockComponent(String name, String type, Simple
     } else if (name.equals(MockTrendline.TYPE)) {
       return new MockTrendline(editor);
     } else {
-      // TODO(user): add 3rd party mock component proxy here
-      throw new UnsupportedOperationException("unknown component: " + name);
+      boolean isExternal = scd.getComponentExternal(name);
+      if (!isExternal) {
+        throw new UnsupportedOperationException("unknown component: " + name);
+      }
+
+      Image icon = getImageFromPath(scd.getIconName(name), type, editor.getProjectId());
+      String packageName = type.substring(0, type.lastIndexOf("."));
+      try {
+        return new MockVisibleExtension(editor, name, icon, packageName, scd.getMockInfo(name));
+      } catch (Exception e) {
+        throw new RuntimeException(e);
```

Comment:
In what scenario would this exception be thrown, and how can we make it more "user friendly" so app developers can know what went wrong?